cmake/config.cmake

```diff
@@ -1,5 +1,7 @@
 set(CMAKE_BUILD_TYPE RelWithDebInfo)
-set(XGRAMMAR_BUILD_PYTHON_BINDINGS ON)
+if(NOT DEFINED XGRAMMAR_BUILD_PYTHON_BINDINGS)
```

Comment:
Can we just remove the BUILD_PYTHON_BINDINGS flag and keep XGRAMMAR_BUILD_PYTHON_BINDINGS? So we don't need the changes in config.cmake / CMakeLists.txt